src/main/java/policycompass/fcmmanager/controllers/FCMModels.java

```diff
@@ -227,14 +235,20 @@ public static FCMModelDetail updateFCMModel(int id, JSONObject jsonModel, String
 
 		session.beginTransaction();
 		Query qModel;
-		if(adminUserFlag==1)
+
+		//is Admin user or not
+		boolean isUserGods=isGods(userPath);
```

Comment:
There is still an _optional_ optimisation possible here: Only check if user is god if his userPath doesn't equal the retrieved model's userPath.